### PR TITLE
Multi-choice component

### PR DESCRIPTION
**Pull request by juanca**
https://www.pivotaltracker.com/story/show/172866943

## PR upkeep checklist

- [x] Change log entry
- [x] Label(s)
- [x] Assignee(s)
- [x] Deployment URL: https://mavenlink.github.io/design-system/multi-choice-component
- [x] (When ready for review) Reviewer(s)

### CODE REVIEW

File: src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React, {
```
Comment (cincospenguinos) [inline src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx:1]: Do we want to inline this or is the line too long?

File: src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.test.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import {
+  render,
+  screen,
+} from '@testing-library/react';
+import CustomFieldInputMultipleChoice from './custom-field-input-multiple-choice.jsx';
+
+describe('<CustomFieldInputMultipleChoice>', () => {
```
Comment (cincospenguinos) [inline src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.test.jsx:9]: This is usually the path to the component--in this case `components/custom-field-input-multiple-choice/custom-field-input-multiple-choice`. Is that what we want to have instead?

File: src/components/section-row/section-row.css [hunk 3]
```diff
@@ -4,6 +4,10 @@
   display: flex;
 }
 
+.section-row > * {
+  flex-basis: 200px;
```
Comment (bennapp) [inline src/components/section-row/section-row.css:8]: Do we want this for all elements after section row, or does this more pertain to usages with custom fields?

Comment (juanca) [reply src/components/section-row/section-row.css:8]: As of now, we know we only want "form fields" in section rows and they need to be aligned in a "grid" through spacing.

File: src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.test.jsx (hunk 2 above)
Comment (juanca) [reply src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.test.jsx:9]: I just checked and it seems we're all over the place in our MDS repo. Hmmm... 

I think we can do either, as long as we're reasonable. 

- Use `<ComponentName>` to indicate that the test suite is for a component
- Use `ComponentName component` to indicate the test suite is for a component
- Use `src/components/component-name` to indicate the test suite is for a component

In contrast to linters:

- Use `ruleName linter` to indicate that the test suite is for a linter
- Use `src/linter/rule-name` to indicate the test suite is for a linter

File: src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx (hunk 1 above)
Comment (juanca) [reply src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx:1]: I'm prefer multi-lining things to be a bit more git friendly! if we add or remove named imports then we don't mess with the other lines.

File: src/components/tag-list/tag-list.jsx [hunk 4]
```diff
@@ -53,7 +53,8 @@ export default function TagList(props) {
 
   return (
     <div // eslint-disable-line jsx-a11y/interactive-supports-focus
-      className={styles['tag-list']}
+      className={props.classContainer}
```
Comment (juanca) [inline src/components/tag-list/tag-list.jsx:56]:
```suggestion
      className={props.classNameContainer}
```

```suggestion
      className={props.classContainer}
```

```suggestion
      className={props.cssContainer}
```

```suggestion
      className={props.className}
```

File: src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import PropTypes from 'prop-types';
+import Icon from '../icon/icon.jsx';
+import iconCaretDown from '../../svgs/icon-caret-down.svg';
+import iconCaretDownDisabled from '../../svgs/icon-caret-down-disabled.svg';
+import TagList from '../tag-list/tag-list.jsx';
+import Tag from '../tag/tag.jsx';
+import styles from './custom-field-input-multiple-choice.css';
+
+function CustomFieldInputMultipleChoice(props) {
+  const refs = props.value.map(() => createRef());
```
Comment (juanca) [inline src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.jsx:13]: I think this is better?

```suggestion
  const refs = props.value.map(() => useRef());
```

File: src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.test.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      expect(screen.getByText('Choice 1').parentElement).toHaveAttribute('id', '123-1');
+      expect(screen.getByText('Choice 2').parentElement).toHaveAttribute('id', '123-2');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render((<CustomFieldInputMultipleChoice {...requiredProps} label="Unique label" />));
+
+      expect(screen.getByLabelText('Unique label')).toBeDefined();
```
Comment (juanca) [inline src/components/custom-field-input-multiple-choice/custom-field-input-multiple-choice.test.jsx:42]:
```suggestion
      expect(screen.getByLabelText('Unique label')).toBeInTheDocument();
```

**Comment by juanca** [comment]
LGT mob review

**Pull request merged**